### PR TITLE
Add LoRA fine-tuning demo notebook

### PR DESCRIPTION
**Pull request by andrefoo**
## Description
This PR adds a basic Jupyter notebook demonstrating how to perform LoRA (Low-Rank Adaptation) fine-tuning using Fireworks AI. The notebook provides a step-by-step guide for customizing language models efficiently using customer support data as an example.

### Features
- Complete setup instructions including prerequisites
- Dataset preparation and formatting
- LoRA fine-tuning with both default and custom parameters
- Model deployment and testing
- Resource cleanup guidance

### Key Components
- Installation and credential setup
- Dataset loading and formatting for Fireworks' API
- Fine-tuning job creation and monitoring
- Model deployment and inference testing
- Resource management (deployment/undeployment)

### Technical Details
- Uses `llama-v3p1-8b-instruct` as the base model
- Demonstrates LoRA parameter configuration (rank, learning rate, epochs)
- Includes error handling and troubleshooting guidance
- Compatible with Google Colab environment

## Testing Done
- ✅ Notebook runs end-to-end in Google Colab
- ✅ All code cells execute successfully
- ✅ Resource cleanup steps verified

## Documentation
- Added detailed comments and explanations throughout the notebook
- Included links to relevant documentation
- Added parameter explanations and best practices

### CODE REVIEW

File: learn/lora-finetuning/LoRA_Finetuning_Demo.ipynb [hunk 1]
```diff
@@ -0,0 +1,1875 @@
...
+        "ACCOUNT_ID = \"YOUR_ACCOUNT_ID\"  # Replace with your account ID\n",
+        "HEADERS = {\"Authorization\": f\"Bearer {API_KEY}\"}\n",
+        "HEADERS_WITH_CONTENT_TYPE = {\n",
+        "    \"Authorization\": f\"Bearer {API_KEY}\",\n",
+        "    \"Content-Type\": \"application/json\"\n",
+        "}\n",
+        "\n",
+        "# Project configuration\n",
+        "BASE_URL = f\"https://api.fireworks.ai/v1/accounts/{ACCOUNT_ID}\"\n",
+        "BASE_MODEL = \"llama-v3p1-8b-instruct\"\n",
```
Comment (dphuang2) [inline learn/lora-finetuning/LoRA_Finetuning_Demo.ipynb:201]: I kept getting the following error when running `firectl create sftj`.

```
2025/02/25 22:08:33 Failed to execute: error creating supervised fine-tuning job: rpc error: code = NotFound desc = error fetching resource: Model accounts/dhuang-97a80b/models/llama-v3p1-8b-instruct not found
```

Until I set the BASE_MODEL param to `accounts/fireworks/models/llama-v3p1-70b-instruct`.

**Comment by Hecate0821** [comment]
Closing this public-side PR to avoid diverging fw-ai/cookbook from the Fireworks monorepo source of truth. Cookbook changes should be staged under public-repos/cookbook in fw-ai/fireworks and then promoted by the public repo promotion workflow.
